pmd-test/src/main/java/net/sourceforge/pmd/AbstractRuleSetFactoryTest.java

```diff
@@ -230,6 +231,36 @@ void testXmlSchema() throws Exception {
         assertTrue(allValid, "All XML must parse without producing validation messages.");
     }
 
+    @Test
+    void verifyCorrectXmlEncoding() throws Exception {
```

Comment:
rather than a test, this seems like something we can create a PMD rule for and have it enabled in our dogfooding.

Reply:
I've just realized that the XML declaration data in unobtainable from our AST, so this rule could not be implemented… so I extended our XML language and created the rule in #4592 :)